doc/guides/west/manifest.rst

```diff
@@ -788,6 +787,120 @@ In this case, both ``groupA`` and ``groupB`` are disabled.
 
 Therefore, projects ``foo`` and ``bar`` are both inactive.
 
+.. _west-group-filter-imports:
+
+Group Filters and Imports
+=========================
+
+This section provides a simplified description of how the ``manifest:
+group-filter:`` value behaves when combined with :ref:`west-manifest-import`.
+For complete details, see :ref:`west-manifest-formal`.
+
+.. warning::
+
+   The semantics are different in in west v0.9.x. There, only the top level
+   manifest file's ``group-filter`` had any effect. This was changed in west
+   0.10.0 to the behavior described below.
+
+   Combining ``group-filter`` with ``import`` in west v0.9.x is discouraged.
```

Comment:
Maybe simplify this.
Not sure readers care so much about the details, but should simply be warned about west `0.9.0` and import.

Just like we don't warn about the west 0.7.x import and missing `self:path` in manifest (I helped someone on slack just yesterday that faced this issue)

```suggestion
   In west v0.9.x only the top level ``group-filter`` has effect, therefore
   combining ``group-filter`` with ``import`` in west v0.9.x is discouraged and west v0.10.0 or newer should be used.
```

Reply:
Changed to:

```
   The below semantics apply to west v0.10.0 and later. West v0.9.x semantics
   are different, and combining ``group-filter`` with ``import`` in west v0.9.x
   is discouraged.
```